lib/bankaccounts/checking_account.rb

```diff
@@ -0,0 +1,47 @@
+module BankAccounts
+
+  class CheckingAccount < Account
+      # attr_accessor :check_count
+
+    def initialize(id, initial_balance)
+      super
+      @check_count = 0
+    end
+
+    def withdraw(amount)
+      # does not allow the account to go negative.
+      # will output a warning message and return the original un-modified balance
+      if amount > @balance
+        puts "You can't withdraw what you don't have..."
+      else
+      # the input amount gets taken out of the account as result of an ATM transaction.
+        @balance = @balance - (amount + 1)
```

Comment:
You may consider extracting the idea of a transaction fee, so that your code is more readable.